File: Java/Palindrome.java
import java.util.Scanner;

public class Palindrome {
  public static void main(String[] args) {
    try (Scanner scanner = new Scanner(System.in)) { // Takes in user input
      System.out.println("Please enter your input: ");
      String input = scanner.nextLine();
      if (isPalindrome(input)) {
        System.out.println("Your input is a palindrome");
      } else {
        System.out.println("Your input is NOT a palindrome");
      }
    }
  }

  public static boolean isPalindrome(String input) {
    StringBuilder reverse = new StringBuilder(input.toLowerCase()).reverse();
    return input.toLowerCase().equals(reverse.toString());
  }
}

File: Javascript/palindrome.js
const input = "aa";

function isPalindrome(input) {
  if (!input || input == "") {
    return false; // If input is null
  }
  let front = 0;
  let back = input.length - 1;

  while (back - front >= 2) {
    if (input[front] != input[back]) {
      return false;
    }
    front++;
    back--;
  }
  return true;
}

console.log(isPalindrome(input));
